pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -0,0 +1,148 @@
+			BeforeEach(func() {
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+				handler.EXPECT().LinkByName(ifaceName).Return(podLink, nil)
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, fmt.Errorf("failed to compute addresses"))
+				})
+
+				It("should fail to read the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+				})
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, nil)
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMIpv6CIDR).Return(ipv6GwAddr, vmIPv6Addr, fmt.Errorf("failed to compute addresses"))
+					handler.EXPECT().ParseAddr(ipv4GwAddr).Return(ipv4GwIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv4Addr).Return(ipv4VmIP, nil)
+					handler.EXPECT().IsIpv6Enabled(ifaceName).Return(true, nil)
+
+				})
+
+				It("should fail to read the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+				})
+			})
+
+			When("the pod interface only has an IPv4 address", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, nil)
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMIpv6CIDR).Return(ipv6GwAddr, vmIPv6Addr, nil).Times(0)
+					handler.EXPECT().ParseAddr(ipv4GwAddr).Return(ipv4GwIP, nil)
+					handler.EXPECT().ParseAddr(ipv6GwAddr).Return(ipv6GwIP, nil).Times(0)
+					handler.EXPECT().ParseAddr(vmIPv4Addr).Return(ipv4VmIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv6Addr).Return(ipv6VmIP, nil).Times(0)
+					handler.EXPECT().IsIpv6Enabled(ifaceName).Return(false, nil)
+				})
+
+				It("should succeed reading the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+					Expect(masqueradeConfigurator.podNicLink).To(Equal(podLink))
+				})
+			})
+
+			When("the pod interface has both IPv4 and IPv6 addresses", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, nil)
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMIpv6CIDR).Return(ipv6GwAddr, vmIPv6Addr, nil)
+					handler.EXPECT().ParseAddr(ipv4GwAddr).Return(ipv4GwIP, nil)
+					handler.EXPECT().ParseAddr(ipv6GwAddr).Return(ipv6GwIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv4Addr).Return(ipv4VmIP, nil)
+					handler.EXPECT().ParseAddr(vmIPv6Addr).Return(ipv6VmIP, nil)
+					handler.EXPECT().IsIpv6Enabled(ifaceName).Return(true, nil)
+				})
+
+				It("should succeed reading the pod link info", func() {
```

Comment:
"should succeed reading the pod link dual stack info"

Also check gateways and pod if addresses for dual stack.

Reply:
pod interface addresses ? Why do we care about that ?